boefjes/boefjes/sql/plugin_storage.py

```diff
@@ -41,8 +41,14 @@ def create_boefje(self, boefje: Boefje) -> None:
         self.session.add(boefje_in_db)
 
     def update_boefje(self, boefje_id: str, data: dict) -> None:
+        if not data:
+            return
+
         instance = self._db_boefje_instance_by_id(boefje_id)
 
+        if instance.static:
+            raise CannotUpdateStaticPlugin(boefje_id)
```

Comment:
Seems to be a bit too specific imho. Wouldn't a `NotAllowed` with a good message suffice?